media-libs/avisynthplus/avisynthplus-3.7.0.recipe

```diff
@@ -0,0 +1,59 @@
+SUMMARY="AviSynth+ scripting frameserver"
+DESCRIPTION="A powerful nonlinear scripting language for video."
+HOMEPAGE="https://avs-plus.net"
+COPYRIGHT="2000-2015 Ben Rudiak-Gould, et al., \
+2013-2021 AviSynth+ Project"
+LICENSE="GNU GPL v2"
+REVISION="1"
+SOURCE_URI="https://github.com/AviSynth/AviSynthPlus/archive/v$portVersion.tar.gz"
+SOURCE_FILENAME="avisynthplus-$portVersion.tar.gz"
+SOURCE_DIR="AviSynthPlus-$portVersion"
+CHECKSUM_SHA256="8906d9e46dc90a194413d69b710e3106c0aafddc0c5c62004885d0c3beb79862"
+#PATCHES=""
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64 ?arm ?ppc"
+SECONDARY_ARCHITECTURES="!x86_gcc2 x86"
+
+PROVIDES="
+	avisynthplus$secondaryArchSuffix = $portVersion
+	lib:libavisynth = $portVersion
+	lib:libconvertstacked = $portVersion
+	lib:libtimestretch = $portVersion
+	lib:libimageseq = $portVersion
+	lib:libshibatch = $portVersion
+	"
+REQUIRES="
+    haiku$secondaryArchSuffix
+    devil$secondaryArchSuffix
+    "
+
+BUILD_REQUIRES="
+    devil${secondaryArchSuffix}_devel
+    "
+BUILD_PREREQUIRES="
+    haiku${secondaryArchSuffix}_devel
+    cmd:gcc$secondaryArchSuffix
+    cmd:cmake
+    cmd:make
+    "
+
+PATCH()
+{
+	sed -i '8i#ifdef AVS_HAIKU\n#define system_avs_plugindir "/system/lib/avisynth"\n#else' avs_core/avisynth_conf.h.in
+	sed -i '12i#endif' avs_core/avisynth_conf.h.in
+}
+
+BUILD()
+{
+	mkdir -p avisynth-build
+	cd avisynth-build
+	cmake ../ $cmakeDirArgs -DCMAKE_CXX_FLAGS="-DRELEASE_TARBALL"
```

Comment:
Any reason for -DCMAKE_CXX_FLAGS="-DRELEASE_TARBALL"? Haven't seen it passing by in other builds

Reply:
The initialization of a git repository despite being downloaded from a tarball causes a build error because the internal versioning assumes you're either compiling from a git clone of the repository or compiling from a tarball with no .git directory.  Because the initialized directory doesn't have the necessary data in it for the git method, it fails unless tarball mode is forced.

Reply:
Thanks for clarifying :+1: